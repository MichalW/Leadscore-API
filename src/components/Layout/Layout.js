import PropTypes from 'prop-types';
import React from 'react';
import {Container, Grid} from 'semantic-ui-react';

import Navbar from '../Navbar';

const Layout = ({children}) => (
  <Container>
    <Navbar />

    <Grid centered>
      <Grid.Row>
        <Grid.Column>
          <Container>
            {children}
          </Container>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  </Container>
);

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
