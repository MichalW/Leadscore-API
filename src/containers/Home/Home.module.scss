.wrapper p {
  padding-bottom: 20px;
}
