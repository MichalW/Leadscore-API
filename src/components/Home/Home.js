import PropTypes from 'prop-types';
import React from 'react';
import {Segment} from 'semantic-ui-react';

import Layout from '../Layout';

const Home = ({username}) => (
  <Layout>
    <Segment>
      {`Signed in as ${username}`}
    </Segment>
  </Layout>
);

Home.defaultProps = {
  username: '',
};

Home.propTypes = {
  username: PropTypes.string,
};

export default Home;
