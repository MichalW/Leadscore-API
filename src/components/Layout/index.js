import {connect} from 'react-redux';

import Layout from './Layout';

const mapStateToProps = state => ({
  signedIn: state.authentication.signedIn,
  username: state.authentication.user.username,
});

export default connect(mapStateToProps)(Layout);
