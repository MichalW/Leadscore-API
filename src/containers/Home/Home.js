import PropTypes from 'prop-types';
import React from 'react';
import {Segment} from 'semantic-ui-react';

import Header from '../../components/Header';
import Layout from '../../components/Layout';
import styles from './Home.module.scss';

const Home = ({username}) => (
  <Layout>
    <Segment className={styles.wrapper}>
      <Header title="Home" icon="home" />

      <p>
        {'Hello '}
        <strong>{username}</strong>
      </p>
    </Segment>
  </Layout>
);

Home.defaultProps = {
  username: '',
};

Home.propTypes = {
  username: PropTypes.string,
};

export default Home;
