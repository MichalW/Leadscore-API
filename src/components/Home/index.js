import {connect} from 'react-redux';

import Home from './Home';

const mapStateToProps = state => ({
  username: state.authentication.user.username,
});

export default connect(mapStateToProps)(Home);
