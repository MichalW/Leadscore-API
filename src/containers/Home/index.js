import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import Home from './Home';
import {signOutRequest} from '../../actions/authentication';

const mapStateToProps = (state) => ({
  username: state.authentication.user.username,
});

const mapDispatchToProps = (dispatch) => ({
  signOutRequest: bindActionCreators(signOutRequest, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(Home);
