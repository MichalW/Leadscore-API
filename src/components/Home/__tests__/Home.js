import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';

import Home from '../Home';
import Layout from '../../Layout';

it('should render Home', () => {
  const props = {
    username: 'ExampleUsername',
  };

  const home = shallow(<Home {...props} />);

  expect(home.find(Layout).exists()).toBe(true);
  expect(toJson(home)).toMatchSnapshot();
});
