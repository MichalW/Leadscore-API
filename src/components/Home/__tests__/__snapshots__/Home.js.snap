// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home 1`] = `
<Layout>
  <Segment>
    Signed in as ExampleUsername
  </Segment>
</Layout>
`;
